util/cache/redis.go

```diff
@@ -28,24 +29,37 @@ type redisCache struct {
 	codec      *rediscache.Codec
 }
 
+// if redis server goes down the first client request fails with EOF error
+func doWithRetry(action func() error) error {
+	err := action()
+	if err == io.EOF {
```

Comment:
any danger of thundering herd?

Reply:
I don't think so. EOF happens only if redis got restarted and one retry won't hurt.